app/policies/step_policy.rb

```diff
@@ -0,0 +1,7 @@
+class StepPolicy < Struct.new(:user, :step)
+  User.steps.each do |state|
```

Comment:
A little neat Ruby meta-programming. It will take each of the possible User steps and define a method at runtime. So whereas `User.steps #=> ["stage_1", ..., "stage_5"]` we define, for example, `User#show_stage_1?` which will ask if the User's current stage is greater than or equal (lexicographically, but I purposely chose stage names that would be related lexicographically the same as they are in chronological order for the business logic) to, in this case, `stage_1`. This is used to power the dropdown menu, only linking to a step that the requirements allow for the User to see.